### PR TITLE
Can now change the name of the scratch directory

### LINKED ISSUE #722: Cannot change the name of the scratch directory
Some parts of the code refer to the directory as 'scratch' and others refer to 'scratchdir'.  Does anyone have a strong preference for either one.  This would be a entry in the MTTDefaults section of the .ini file and an option in the command line as --scratch-dir.

### PR DESCRIPTION
**Pull request by DebRez**
The name of the scratch directory can now be changed in the MTTDefaults section of the .ini file using either scratch = or scratchdir = and on the command line using --scratch-dir

Fixes #722

Signed-off-by: Deb Rezanka <rezanka@lanl.gov>

### CODE REVIEW

**Comment by DebRez** [comment]
I need to add NoOptionError exceptions - working on it now

**Comment by DebRez** [comment]
All fixed. I tried all three ways of setting the scratch directory name and they worked.

File: pyclient/pymtt.py [hunk 1]
```diff
@@ -82,7 +82,7 @@
                      help="Specify the DIRECTORY where additional plugins can be found (or comma-delimited list of DIRECTORYs)", metavar="DIRECTORY")
 execGroup.add_argument("--ignore-loadpath-errors", action="store_true", dest="ignoreloadpatherrs", default=False,
                      help="Ignore errors in plugin paths")
-execGroup.add_argument("--scratch-dir", dest="scratchdir", default="./mttscratch",
+execGroup.add_argument("--scratch-dir", dest="scratchdir", default=None,
```
Comment (ribab) [inline pyclient/pymtt.py]: Why is the default set to None?

File: pylib/System/TestDef.py [hunk 2]
```diff
@@ -76,11 +76,6 @@ def __init__(self):
     def setOptions(self, args):
         self.options = vars(args)
         self.args = args
-        # if they want us to clear the scratch, then do so
-        if self.options['clean'] and os.path.isdir(self.options['scratchdir']) :
-            shutil.rmtree(self.options['scratchdir'])
-        # setup the scratch directory
-        _mkdir_recursive(self.options['scratchdir'])
```
Comment (ribab) [inline pylib/System/TestDef.py]: This worked before, why remove the creating of scratchdir?

File: pylib/System/TestDef.py [hunk 3]
```diff
@@ -613,6 +608,25 @@ def configTest(self):
...
+        try:    
+            if not self.options['executor']:
+                self.options['executor'] = self.config.get('MTTDefaults', 'executor')
+        except:
+            self.options['executor'] = 'sequential'        
+        # if they want us to clear the scratch, then do so
+        if self.options['clean'] and os.path.isdir(self.options['scratchdir']) :
+            shutil.rmtree(self.options['scratchdir'])
+        # setup the scratch directory
+        _mkdir_recursive(self.options['scratchdir'])
```
Comment (ribab) [inline pylib/System/TestDef.py]: Why move creating of scratchdir here? There seems to be some code that depends on scratchdir being there that executes before this part of the code

**Pull request merged**